packages/react-scripts/scripts/utils/createJestConfig.js

```diff
@@ -24,27 +24,27 @@ module.exports = (resolve, rootDir, srcRoots) => {
   // TODO: I don't know if it's safe or not to just use / as path separator
   // in Jest configs. We need help from somebody with Windows to determine this.
   const config = {
-    collectCoverageFrom: ['src/**/*.{js,jsx,mjs}'],
+    collectCoverageFrom: ['src/**/*.{js,jsx,mjs,ts,tsx}'],
     setupFiles: [resolve('config/polyfills.js')],
     setupTestFrameworkScriptFile: setupTestsFile,
     testMatch: [
-      '**/__tests__/**/*.{js,jsx,mjs}',
-      '**/?(*.)(spec|test).{js,jsx,mjs}',
+      '**/__tests__/**/*.{js,jsx,mjs,ts,tsx}',
+      '**/?(*.)(spec|test).{js,jsx,mjs,ts,tsx}',
     ],
     // where to search for files/tests
     roots: srcRoots.map(toRelRootDir),
     testEnvironment: 'node',
     testURL: 'http://localhost',
     transform: {
-      '^.+\\.(js|jsx|mjs)$': resolve('config/jest/babelTransform.js'),
+      '^.+\\.(js|jsx|mjs|tsx?)$': resolve('config/jest/babelTransform.js'),
```

Comment:
Just leaving a note but the `config/jest/babelTransform.js` file would need to be updated to right now its:

https://github.com/brunolemos/create-react-app/blob/f33e0ca7f7fcb7474cb87ce0e73e74c6332a53e7/packages/react-scripts/config/jest/babelTransform.js#L13-L18
```js
module.exports = babelJest.createTransformer({
  presets: [require.resolve('babel-preset-react-app')],
  // @remove-on-eject-begin
  babelrc: false,
  // @remove-on-eject-end
});
```

Should probably be the similar as the webpack setup:
```js
module.exports = babelJest.createTransformer({
  presets: [require.resolve('babel-preset-react-app')],
  require.resolve('@babel/preset-typescript')  
  // @remove-on-eject-begin
  babelrc: false,
  // @remove-on-eject-end
});
```

I haven't tried using jest at all so I'm not sure if this would work but leaving it as a reminder for me as well :).

Reply:
@ksrb no need because `@babel/preset-typescript` is being added automatically by `babel-preset-react-app` on this fork

Reply:
Oh right nvm :)